src/main/java/org/jenkinsci/plugins/plaincredentials/impl/FileCredentialsImpl.java

```diff
@@ -67,6 +67,12 @@
             this.fileName = fileName;
             this.data = Base64.decodeBase64(data);
         }
+        if (this.fileName == null || this.fileName.isEmpty()) {
+            throw new IllegalArgumentException(
+                    String.format("No FileName was provided or resolved. " +
+                            "Input file item was %s and input file name was %s.", file.toString(), fileName)
```

Comment:
Ii can be different, I mean you can go there because `this.fileName = name.replaceFirst("^.+[/\\\\]", "");`gets blank but not because the entry fileName (the method attribute) which can be blank or not.